File: code_golf/java/passwordify_string.java
/*
 * 7/3/2019
 * 
 * Passwordify the string
 * 
 * https://codegolf.stackexchange.com/questions/76486/passwordify-the-string
 * 
 * Take a string as input. This string will only contain uppercase letters, lowercase letters, 
 * digits, and spaces.
 * 
 * You must replace all spaces with underscores and move all numbers to the end of the string in
 * the order that they appear from left to right. Then, for every letter in the string, randomly
 * change it to uppercase or lowercase.
 * 
 */
public class PasswordifyString {
	
	public static void main(String[] args) {
		String[] inputStrings = {"1pass 2word3 ", "Jessie Edmisten", "12asdfg34qwert56zx", "Hello world", "pa55w0rd", "14 35", "0971", " "};
		
		for (String inputStr : inputStrings) {
			System.out.println(inputStr);
			System.out.println(passwordifyString(inputStr) + "\n");
		}
	}
	
	public static String passwordifyString(String inputStr) {
		String digitStr = ""; 
		String randStr = "";
		
		inputStr = inputStr.replace(" ", "_");
		
		// Create string of digits from input string in the order they appeared from left to right.
		for (int i = 0; i < inputStr.length(); i++) {
			if (Character.isDigit(inputStr.charAt(i))) {
				digitStr += inputStr.charAt(i);
			}
		}
		
		// Remove digits from input string.
		inputStr = inputStr.replaceAll("([0-9])", "");
		
		// Randomly make letters in input string upper or lower case.
		for (int i = 0; i < inputStr.length(); i++) {
			int randValue = (int) Math.round(Math.random());
			if (randValue == 0) {
				randStr += Character.toLowerCase(inputStr.charAt(i));
			} else {
				randStr += Character.toUpperCase(inputStr.charAt(i));
			}
		}
		
		return randStr + digitStr;
	}

}

File: code_golf/C#/PasswordifyString.cs
/*
 * 2/7/2021
 * 
 * Passwordify the string
 * 
 * https://codegolf.stackexchange.com/questions/76486/passwordify-the-string
 * 
 * Take a string as input. This string will only contain uppercase letters, lowercase letters, 
 * digits, and spaces.
 * 
 * You must replace all spaces with underscores and move all numbers to the end of the string in
 * the order that they appear from left to right. Then, for every letter in the string, randomly
 * change it to uppercase or lowercase.
 * 
 */

using System;

namespace PasswordifyString2
{
    class Program
    {
        static void Main(string[] args)
        {
            string[] words = {"Hello world", "pa55 w0rd", "14 35", "0971", " ", "1Jessie2 3Edmisten4",
                              "1pass 2word3", "12asdfg34qwert56zx"};

            foreach (string word in words)
            {
                Console.WriteLine(word);
                Console.WriteLine(passwordify(word));
                Console.WriteLine();
            }
        }

        static string passwordify(string word)
        {
            word = word.Replace(" ", "_");

            string digitStr = "";
            string charStr = "";
            Random rand = new Random();
            for (int i = 0; i < word.Length; i++)
            {
                if (Char.IsDigit(word[i]))
                {
                    digitStr += word[i];
                }
                else
                {
                    charStr += (rand.Next(0, 2) == 0) ? Char.ToUpper(word[i]) : Char.ToLower(word[i]);
                }
            }

            return charStr + digitStr;
        }
    }
}